tests/prng_tools/main.c

```diff
@@ -0,0 +1,112 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "shell.h"
+#include "random.h"
+
+static int cmd_distributions(int argc, char **argv);
+
+/**
+ * @brief   List of command for this application.
+ */
+static const shell_command_t shell_commands[] = {
+    { "distributions", "run distributions test", cmd_distributions },
+    { NULL, NULL, NULL }
+};
+
+/**
+ * @brief   Seed for initializing random module
+ */
+static uint32_t seed = 0;
+
+/**
+ * @brief   Helper for printing a vertical distribution
+ */
+static void print_distribution(uint32_t* values, uint8_t bins)
+{
+    printf("For %d bit samples:\n", bins);
+
+    /* sum the total number of ones */
+    uint64_t total = 0;
+
+    for (int i = 0; i < bins; i++) {
+        total = total + values[i];
+    }
+
+    /* if total is zero, it would yield a division by zero */
```

Comment:
IMHO `assert` would be a program error, where this case is a likely scenario (for a bad RNG).